### ISSUE TITLE
@Provides method is not accessible with "protected"

### ISSUE THREAD

**Issue by coldwised** [evant/kotlin-inject#307]
Hello. In this example, I think that the "provideFoo()" function should be annotated with the @Provide annotation. If so, then there will be an error "@Provides method is not accessible". It works only with public functions. Can I somehow close access to the function for all but child components? It would make sense if the child components would implement the parent component, which we pass to the child component's constructor, and have access to protected properties and functions. 
![image](https://github.com/evant/kotlin-inject/assets/71338871/625c3821-02b2-4c73-87ec-94f60432c74c)

**Comment by evant** [comment]
Yes this looks like a documentation bug. As for allowing protected provides, it is something I would like to support though haven't worked through all the implementation details. It would require generating bridge methods for the child component to call.
